File: naive_primes.c
#include <stdio.h>
#include <stdlib.h>

#define BUFFER_SIZE 10000

static inline int is_prime(unsigned short n) {
    unsigned short i;
    for (i = 2; i < n; i++) {
        if (n % i == 0) {
            return 0;
        }
    }
    return 1;
}

int main() {
    unsigned short i;
    unsigned short* buffer = malloc(BUFFER_SIZE*sizeof(short));
    unsigned short count = 0;
    for (i = 2; i <= 65000; i++) {
        if (is_prime(i)) {
            if (count >= BUFFER_SIZE) {
                fprintf(stderr, "Error: buffer overflow\n");
                return 1;
            }
            buffer[count++] = i;
        }
    }
    printf("count: %d, last prime: %d", count, buffer[count-1]);
    return 0;
}

File: naive_primes.py
BUFFER_SIZE = 10000

def is_prime(n):
    for i in range(2, n):
        if n % i == 0:
            return False
    return True

def main():
    buffer = []
    count = 0
    for i in range(2, 65001):
        if is_prime(i):
            buffer.append(i)
            count += 1
    print("count:", count, ", last prime:", buffer[-1])

if __name__ == "__main__":
    main()
